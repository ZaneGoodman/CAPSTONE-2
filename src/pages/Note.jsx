import React from "react";

const Note = ({ note }) => {
  return <p>{note}</p>;
};

export default Note;
