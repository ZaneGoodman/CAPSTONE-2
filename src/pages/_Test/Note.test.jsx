import React from "react";
import { render } from "@testing-library/react";

import AuthProvider from "../../provider/authProvider";
import Note from "../Note";
//Smoke test
test("renders Note component with authentication", async () => {
  render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <Note note="Test Note" />
    </AuthProvider>
  );
});

//Snapshot test  <MemoryRouter>
it("Note matches snapshot", async () => {
  const { asFragment } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <Note note="Test Note" />
    </AuthProvider>
  );
  await expect(asFragment()).toMatchSnapshot();
});
