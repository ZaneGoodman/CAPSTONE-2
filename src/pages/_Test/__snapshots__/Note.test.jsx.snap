// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note matches snapshot 1`] = `
<DocumentFragment>
  <p>
    Test Note
  </p>
</DocumentFragment>
`;
